src/memcache/handler.cpp

```diff
@@ -195,15 +219,18 @@ std::unique_ptr<cybozu::tcp_socket> handler::make_memcache_socket(int s) {
 std::unique_ptr<cybozu::tcp_socket> handler::make_repl_socket(int s) {
     if( m_slaves.size() == MAX_SLAVES )
         return nullptr;
-    std::unique_ptr<repl_socket> t(
-        new repl_socket(s, g_config.repl_bufsize(), m_finder) );
-    repl_socket* pt = t.get();
-    m_slaves.push_back(pt);
-    m_syncer.add_request(
-        std::unique_ptr<sync_request>(
-            new sync_request([this,pt]{ m_new_slaves.push_back(pt); })
-            ));
-    return std::move(t);
```

Comment:
The reason why we cannot change this line as `return t;` is described here:
https://stackoverflow.com/questions/22018115/converting-stdunique-ptrderived-to-stdunique-ptrbase#

Specifically, gcc-4.8.x would fail to compile if we change this line to `return t;`.
gcc-5.x compiles it nicely.